scrapy/extensions/feedexport.py

```diff
@@ -341,14 +345,16 @@ def close_spider(self, spider):
         return defer.DeferredList(deferred_list) if deferred_list else None
 
     def _close_slot(self, slot, spider):
-        if not slot.itemcount and not slot.store_empty:
+        if not self.batches[slot.uri_template].updated_once and not slot.store_empty:
             # We need to call slot.storage.store nonetheless to get the file
             # properly closed.
             return defer.maybeDeferred(slot.storage.store, slot.file)
         slot.finish_exporting()
-        logfmt = "%s %%(format)s feed (%%(itemcount)d items) in: %%(uri)s"
+        logfmt = "%s %%(format)s feed in: %%(uri)s, batch state: %%(batch_state)s"
+        batch_state = self.batches[slot.uri_template].get_batch_state()
+        batch_state_str = ", ".join("{}={}".format(k, v) for k, v in batch_state.items())
         log_args = {'format': slot.format,
-                    'itemcount': slot.itemcount,
+                    'batch_state': batch_state_str,
```

Comment:
I don’t think we need to modify the logged message. If the goal is for the logs to explain why the batch was triggered, it probably makes sense to log a message about that separately. This message is about informing users that a file with N items has been stored in a given location, regardless of the reason (time, number of items, etc.).

Reply:
I thought that using that log message to convey the batch state will serve both points: why batch was triggered and what was the last state when it was closed. Though I see your point and I will try that.